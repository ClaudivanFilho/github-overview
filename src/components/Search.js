import React, { Component } from 'react'
import PropTypes from 'prop-types'

import RepositoryContext from '../contexts/RepositoryContext'
import SearchIcon from '../images/glass.svg'

class Search extends Component {
  static propTypes = {
    originalRepositories: PropTypes.arrayOf(PropTypes.object),
    updateRepositories: PropTypes.func,
  }

  state = {
    inputText: '',
  }

  handleTextChange = (evt) => {
    this.setState({
      inputText: evt.target.value,
    })
    this.handleSearch(evt.target.value)
  }

  handleSearch = (text) => {
    const searchText = text || this.state.inputText
    this.props.updateRepositories(this.props.originalRepositories.filter(rep => (
      rep.nameWithOwner.toLowerCase().includes(searchText.toLowerCase())
    )))
  }

  render() {
    return (
      <div className="flex items-center">
        <input
          id="search"
          placeholder="Search..."
          className="go-search-input input-reset ba b--black-20 db w-100 bg-white br2"
          type="text"
          value={this.state.inputText}
          onChange={this.handleTextChange}
        />
        <img className="pl2 absolute" src={SearchIcon} alt="s-icon" width="20" />
      </div>
    )
  }
}

export default function ComponentWithContext(props) {
  return (
    <RepositoryContext.Consumer>
      {({ originalRepositories, updateRepositories }) => (
        <Search
          {...props}
          originalRepositories={originalRepositories}
          updateRepositories={updateRepositories}
        />
      )}
    </RepositoryContext.Consumer>
  )
}
