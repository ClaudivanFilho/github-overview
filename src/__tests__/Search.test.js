import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import Search from '../components/Search'

describe('Test Search component', () => {
  let mountedComponent = null

  beforeAll(() => {
    mountedComponent = mount(
      <Search />
    )
  })

  it('Renders correctly', () => {
    expect(toJson(mountedComponent)).toMatchSnapshot()
  })

  it('Component should be defined', () => {
    expect(mountedComponent).toBeDefined()
  })

  it('Must find an input component', () => {
    expect(mountedComponent.find('input').exists()).toBe(true)
  })

  it('Must find the search icon', () => {
    expect(mountedComponent.find('img').exists()).toBe(true)
  })
})
