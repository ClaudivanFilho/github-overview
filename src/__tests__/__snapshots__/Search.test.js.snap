// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search component Renders correctly 1`] = `
<ComponentWithContext>
  <Search
    originalRepositories={Array []}
    updateRepositories={[Function]}
  >
    <div
      className="flex items-center w-100 w-auto-m ma2 ma0-m"
    >
      <input
        className="input-reset ba b--silver db w-100 bg-white br2 pl8"
        id="search"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
      <img
        alt="s-icon"
        className="pl3 absolute"
        src="glass.svg"
        width="20"
      />
    </div>
  </Search>
</ComponentWithContext>
`;
